vowpalwabbit/cb_explore.cc

```diff
@@ -353,10 +353,10 @@ base_learner* cb_explore_setup(arguments& arg)
   base_learner* base = setup_base(arg);
   data->cbcs.scorer = arg.all->scorer;
 
-  learner<cb_explore>* l;
+  learner<cb_explore,example>* l;
   if (arg.vm.count("cover"))
   {
-    data->cs = arg.all->cost_sensitive;
+    data->cs = decltype(data->cs)(arg.all->cost_sensitive);
```

Comment:
why do you need the decltype?

Reply:
arg.all->cost_sensitive erases multilabel vs. singlelabel type information so data->cs needs to recover it.  

TODO: Use type addition function which inverts setup_base to safely add type information.

Reply:
Changed.
1) Safe cast with type check to singleline learner.
2) explicit type instead of decltype for clarity.